### PR TITLE
Fix economy mismatch in Request loading

### LINKED ISSUE #3866: Worker request not saved in savegame
Bug reported on forum see https://www.widelands.org/forum/post/32641/

to reproduce best use a minimal start condition. Create 1 or 2 buildings without proper workers where no tools are available as well. save the game. close and reload the saved game. build a toolsmithy. No tools are built although the smithy could  work.

### PR DESCRIPTION
**Pull request by Noordfrees**
Ware requests for Warehouses' PlannedWorkers are added to worker economies on saveloading. This was not corrected later under some circumstances.
Fixes #3866

### CODE REVIEW

**Comment by Niektory** [comment]
Seems to work, I can't reproduce the bug anymore.

**Review by Niektory — COMMENTED** [review]
Hmm... So first a wrong request is created in `MapBuildingdataPacket::read_warehouse()`, and then it has to be corrected in `Request::read()`. Quite a roundabout way... Probably worth a comment in both methods.

Is this safe? Is there any danger doing it this way can trigger an exception in `Request::Request()`?
https://github.com/widelands/widelands/blob/0899f4c082af99bdc9bd55c4195d8fc084931674/src/economy/request.cc#L72-L75
EDIT: Oh, `index` is `0` so I guess it's "safe".

**Comment by Noordfrees** [comment]
Added some comments. There's no risk of the exception being triggered with the initialization values because this check is already being performed in master so that has been working all the time.

File: src/map_io/map_buildingdata_packet.cc [hunk 1]
```diff
@@ -458,6 +458,9 @@ void MapBuildingdataPacket::read_warehouse(Warehouse& warehouse,
 
 				uint32_t nr_requests = fr.unsigned_32();
 				while (nr_requests--) {
+					// We habe no information regarding the index or WareWorker type yet.
```
Comment (Niektory) [inline src/map_io/map_buildingdata_packet.cc]:
```suggestion
					// We have no information regarding the index or WareWorker type yet.
```

**Review by Niektory — APPROVED** [review]
Looks good! Just one typo :)

**Pull request merged**